app.js

```diff
@@ -10,8 +11,194 @@ class KNXApp extends Homey.App {
+            tokens.value_bool = true;
+          }
+      }
+
+      const state = { group_address: groupaddress, interface: args.interface, data_type: args.data_type };
+      await this.receiveTelegramTrigger.trigger(tokens, state);
+    } catch (e) {
+      this.log(e);
+    }
+  }
+
+  /**
+   * Handler for the interface found
+   */
+  async onKNXInterface(knxInterface) {
+    const cards = await this.receiveTelegramTrigger.getArgumentValues()
+    this.registerKNXEventHandlers(cards);
+  }
+
+  registerKNXEventHandlers(cards) {
+    cards.forEach((args) => {
+      if (!this.eventListenerGroupAddresses.includes(`${args.interface.mac}-${args.group_address}`)) {
+        this.log('adding event listener', `${args.interface.mac}-${args.group_address}`);
+
+        let knxInterfaceToUse = this.knxInterfaceManager.getKNXInterface(args.interface.mac);
+        if (args.interface.mac === 'any') {
+          const availableInterfaces = this.knxInterfaceManager.getKNXInterfaceList();
+          const macs = Object.keys(availableInterfaces);
+          knxInterfaceToUse = macs.length > 0 ? availableInterfaces[macs[0]] : null;
+        }
+
+        if (!knxInterfaceToUse) {
+          this.error('Interface not found');
+        } else {
+          this.eventListenerGroupAddresses.push(`${args.interface.mac}-${args.group_address}`);
+          // Store the event listener so we can remove it later
+          knxInterfaceToUse.addKNXEventListener(args.group_address, this.onKNXEvent.bind(this, args));
+        }
+      }
+    });
+
+    this.eventListenerGroupAddresses.forEach((eventListernerId) => {
+      if (cards.filter((args) => `${args.interface}-${args.group_address}` === eventListernerId).length === 0) {
+        // TODO: figuure out how to remove the event listener, because we need the correct event listener
+        // knxInterfaceToUse.removeKNXEventListener(groupAddress, this.onKNXEvent.bind( this, args));
+      }
+    });
+  }
+
+  async interfaceAutocomplete(query, args) {
+    const results = this.knxInterfaceManager.getSimpleInterfaceList();
+    results.push({ name: '[any]', mac: 'any', ip: 'any' });
+
+    // filter based on the query
+    return results.filter((result) => (
+      result.name.toLowerCase().includes(query.toLowerCase())
+      || result.mac.toLowerCase().includes(query.toLowerCase())
+      || result.ip.toLowerCase().includes(query.toLowerCase())
+    ));
+  }
+
+  async datatypeAutocomplete(query, args) {
+    return this.availableKnxDataTypes.filter((r) => r.name.includes(query.toUpperCase()));
+  }
+
+  async readKNXTelegram(args, state) {
+    try {
+      let knxInterfaceToUse = this.knxInterfaceManager.getKNXInterface(args.interface.mac);
+      if (args.interface.mac === 'any') {
+        const availableInterfaces = this.knxInterfaceManager.getKNXInterfaceList();
+        const macs = Object.keys(availableInterfaces);
+        knxInterfaceToUse = macs.length > 0 ? availableInterfaces[macs[0]] : null;
+      }
+
```

Comment:
Please do argument validation just as you did in `readKNXTelegram`
If you throw the error make sure its not caught by your try catch but shown to the user